### PR TITLE
[Homework #17] - Class-and-functional-components by Brisa Rojas

### PR DESCRIPTION
**Pull request by bytheseaside**
Hey Ilya! Here's my homework 17. 
I'm sorry about the delay

Let me know if I have to change something

### CODE REVIEW

**Review by ilya-ischenko — COMMENTED** [review]
It looks good to me.

I have some suggestions how to improve it:

1. Add loading state when add new tile button was clicked. Network connection can be slow and loading takes time, so user should have UI feedback what's going on.

2. Refresh icon on tile should rotate only when image is loading. I mean when you hover on tile, refresh icon will be static, then after click it will be rotating while image will be fetching

3.Add cursor pointer for add new tile button.

4. Add loading state for refresh all functionality

5. Create "components" folder and move dedicated components to this folder, it will be more flexible and easier to manage it

6. Add try/catch functionality. You can just add "alert" message to notify user that image was not fetched

This improvements is on your own, let me know your decision

**Comment by bytheseaside** [comment]
Hi Ilya, I'll work on your suggestions

**Comment by bytheseaside** [comment]
I implemented the loading modifications, but I'm having trouble with the catching error part. Could you give me any suggestions for it? I commited and pushed the code so you can check it

**Review by ilya-ischenko — CHANGES_REQUESTED** [review]
Looks good in overall, but need to fix some issues:

1.[ Read this one](https://en.reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous), please. Pay attention where you use this.state in setState and fix it according to documentation.

2. Components file should start from upper case letter and be in CamelCase, in this way we can emphasize that this files are components 

One tip: if you want to return object with help of arrow function, you can use this one: `() => ({ key: value })` instead of `() => { return { key: value } }`

File: homeworks/brisa-rojas/17-class-and-functional-components/src/App.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  async getNewAvatarImage() {
+    return fetch('https://tinyfac.es/api/data?limit=1')
+      .then((response) => response.json())
+      .then((data) => data[0].url);
+  }
+
+  async refreshAvatarImage(avatarId) {
+    let onLoadAvatar = this.state.avatarLoading;
+    onLoadAvatar[avatarId] = true;
```
Comment (ilya-ischenko) [inline homeworks/brisa-rojas/17-class-and-functional-components/src/App.js:39]: Do not you change this.state.avatarLoading directly???

File: homeworks/brisa-rojas/17-class-and-functional-components/src/App.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    let newAvatarImages = this.state.avatarImages.map((avatarImage, index) => {
+      if (index === avatarId) {
+        return newAvatarURL;
+      } else {
+        return avatarImage;
+      }
+    });
+    this.setState({ avatarImages: newAvatarImages });
+    onLoadAvatar[avatarId] = false;
```
Comment (ilya-ischenko) [inline homeworks/brisa-rojas/17-class-and-functional-components/src/App.js:52]: Do not you change this.state.avatarLoading directly???

File: homeworks/brisa-rojas/17-class-and-functional-components/src/components/avatar-tile.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import refreshIcon from '../refresh-icon.svg';
+
+class AvatarTile extends React.Component {
+  constructor(props) {
+    super(props);
+  }
```
Comment (ilya-ischenko) [inline homeworks/brisa-rojas/17-class-and-functional-components/src/components/avatar-tile.js:7]: Unless constructor

**Review by ilya-ischenko — APPROVED** [review]
Well done!
Approved.

**Pull request merged**